metals/src/main/scala/scala/meta/internal/metals/PostCompileCache.scala

```diff
@@ -0,0 +1,60 @@
+package scala.meta.internal.metals
+
+import java.util.Collections.singletonList
+import java.util.concurrent.CompletableFuture
+import java.util.concurrent.ConcurrentHashMap
+import java.util.concurrent.atomic.AtomicBoolean
+import ch.epfl.scala.{bsp4j => b}
+import scala.meta.internal.metals.MetalsEnrichments._
+
+final class PostCompileCache(buildServer: () => Option[BuildServerConnection]) {
```

Comment:
Would `MainClasses` be an appropriate name for this index?

Reply:
the test classes will be collected here as well, so it would have to be renamed to something more general anyway.
Could also be something like `JvmEntryPoints` or something like that - what do you think?

Reply:
`BuildTargetClasses`? It would represent the main/test classes given a build target

Reply:
given that name I would expect it to contain any kind of class from a build target. Maybe `BuildTargetEntryPoints` ?

@tgodzik what do you think? This class will store every runnable main/test class per build target.

Reply:
There would be methods `BuildTargetClasses.{mainClasses,testClasses}` to clarify what kinds of classes the index contains.

Reply:
consider me convinced ;)